src/strategies.jl

```diff
@@ -124,55 +124,89 @@ end
 
 # ---------------------------------------------------------------------------------
 
+abstract StateUpdater
+
+immutable NoUpdater <: StateUpdater end
+state!(model, su::NoUpdater, obs...) = return
+
+immutable BackpropUpdater <: StateUpdater end
+function state!(model, su, target, input)
```

Comment:
Should this be `state!(model, su::BackpropUpdater, target, input)`? Otherwise we could get silent errors here if some garbage is passed in as `su`